In [1]:
import pandas as pd
from pandas import Series, DataFrame

df_2019_jul = pd.read_csv('../data/nyc_taxi_2019-07.csv',
                usecols=['passenger_count', 
                        'total_amount', 'payment_type'])
df_2019_jul['year'] = 2019

df_2020_jul = pd.read_csv('../data/nyc_taxi_2020-07.csv',
                usecols=['passenger_count', 
                        'total_amount', 'payment_type'])
df_2020_jul['year'] = 2020

df = pd.concat([df_2019_jul, df_2020_jul])

# Beyond 1

Use the `corr` method on `df` to find the correlations among the columns. How would you interpret these results?

In [2]:
df.corr()

passenger_count  payment_type  total_amount      year
passenger_count         1.000000      0.016410      0.014943 -0.049558
payment_type            0.016410      1.000000     -0.138561  0.029277
total_amount            0.014943     -0.138561      1.000000 -0.019706
year                   -0.049558      0.029277     -0.019706  1.000000

There is no significant correlation among any of these columns. The closest that we see is between payment type and total amount — but even there, it's an extremely small, negative correlation. 

(The payment type is actually categorical, so saying that there's a numeric correlation is a bit weird in any event. But given that 1 is credit and 2 is cash, a positive correlation would mean that a higher fare would be more likely to use cash. A negative correlation thus means that a higher fare is *less* likely to be paid in cash, which matches our common sense. But again, the correlation is extremely weak.)

# Beyond 2

Show, with a single command, the difference in descriptive statistics for `total_amount` beween 2019 and 2020. Round values to use no more than 2 digits after the decimal point.

In [3]:
(
    df.loc[df['year'] == 2020, 'total_amount'].describe().round(2) - 
    df.loc[df['year'] == 2019, 'total_amount'].describe().round(2)
)

count   -5510007.00
mean          -0.98
std           -0.75
min           53.20
25%           -0.50
50%           -0.60
75%           -0.75
max        -4672.45
Name: total_amount, dtype: float64

# Beyond 3

If we assume that zero-passenger trips are for delivering packages, how were those affected during the pandemic? Show the proportion of such trips in 2019 vs. 2020.

In [4]:
df.loc[(df['year'] == 2019) & 
       (df['passenger_count'] == 0), 'passenger_count'].count() / df['passenger_count'].count()

0.01666432611802781

In [5]:
df.loc[(df['year'] == 2020) & 
       (df['passenger_count'] == 0), 'passenger_count'].count() / df['passenger_count'].count()

0.0027809994974354953

Turns out that the proportion of zero-person taxi rides shrank as well, from about 1.6% in 2019 to 0.2% in 2020. Honestly, I expected it to be higher during the pandemic!